### PR TITLE
Update CMake code to latest usage

### PR DESCRIPTION
**Pull request by byrnHDF**
Set minimum to 3.12.
Add optional message commands.
Use localized name for ZLIB.
Add OSX architecture option.
Change configuration file install location.
Update sanitizer settings.
Update documentation.

### CODE REVIEW

File: release_notes/USING_HDF4_CMake.txt [hunk 1]
```diff
@@ -29,25 +29,38 @@ Notes: This short instruction is written for users who want to quickly
...
 
 ========================================================================
 I. Preconditions
 ========================================================================
 
-   1. We suggest you obtain the latest CMake for windows from the Kitware
-      web site. The HDF 4.2.x product requires a minimum CMake version
-      of 3.10.1.
+   1. We suggest you obtain the latest CMake for your platform from the Kitware
+      web site. The HDF-4.2.x product requires a minimum CMake version
```
Comment (bmribler) [inline release_notes/USING_HDF4_CMake.txt:41]: Is '-' intentional?  Isn't HDF 4.2.x the way we normally use?

**Review by bmribler — CHANGES_REQUESTED** [review]

File: config/sanitizer/sanitizers.cmake [hunk 2]
```diff
@@ -83,12 +83,21 @@ if(USE_SANITIZER)
...
       else()
         message(FATAL_ERROR "This sanitizer not yet supported in the MSVC environment: ${USE_SANITIZER}")
       endif()
     else()
       message(FATAL_ERROR "USE_SANITIZER is not supported on this platform.")
     endif()
+  elseif(MSVC)
+    if(USE_SANITIZER MATCHES "([Aa]ddress)")
+      message(STATUS "Building with Address sanitizer")
+      append("/fsanitize=address" CMAKE_C_SANITIZER_FLAGS CMAKE_CXX_SANITIZER_FLAGS)
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply config/sanitizer/sanitizers.cmake:96]: windows uses "/"

File: release_notes/INSTALL_CMake.txt [hunk 3]
```diff
@@ -304,7 +306,8 @@ IV. Further considerations
             TGZPATH:STRING="some_location"
          where "some_location" is the URL or full path to the compressed
          file and ext is the type of compression file. Also set CMAKE_BUILD_TYPE
-         to the configuration type during configuration
+         to the configuration type during configuration. See the settings in the
+         config/cmake/cacheinit.cmake file HDF uses for testing.
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply release_notes/INSTALL_CMake.txt:310]: Looks correct after reading it aloud.

File: release_notes/INSTALL_CYGWIN.txt [hunk 4]
```diff
@@ -2,25 +2,26 @@
          HDF4 Build and Install Instructions for Cygwin
 ************************************************************************
 
-This file provides instructions to build, test, and install the HDF4
-library on Cygwin. For further information, see the INSTALL file. Also,
-see the INSTALL_CMake.txt file if building with CMake. 
+This document is a instruction on how to build, test and install HDF4 library on
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply release_notes/INSTALL_CYGWIN.txt:5]: right.

File: release_notes/USING_CMake_Examples.txt [hunk 5]
```diff
@@ -20,15 +20,14 @@ Notes: This short instruction is written for users who want to quickly
 I. Preconditions
 ========================================================================
 
-   1. We suggest you obtain the latest CMake for windows from the Kitware
-      web site. The HDF 4.2.x product requires a minimum CMake version
-      of 3.2.2.
+   1. We suggest you obtain the latest CMake for your platform from the Kitware
+      web site. The HDF5 1.13.x product requires a minimum CMake version
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply release_notes/USING_CMake_Examples.txt:24]: corrected

File: release_notes/USING_CMake_Examples.txt [hunk 6]
```diff
@@ -59,17 +58,19 @@ Default installation process:
            be the same as the value used with the -C command line option.
        The default build configuration is defined to build and use static libraries.
            Shared libraries can be used with the STATIC_ONLY script option set to "NO".
-       Other options can be changed by editing the HDF4_Examples_options.cmake file.
+
+       Other options can be changed by editing the 
+           HDF4_Examples_options.cmake file.
 
        If the defaults are okay, execute from this directory:
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply release_notes/USING_CMake_Examples.txt:65]: this is one of the steps, starting with the directory to create - missing is the direction to cd into the directory.

File: release_notes/USING_CMake_Examples.txt [hunk 7]
```diff
@@ -59,17 +58,19 @@ Default installation process:
...
            ctest -S HDF4_Examples.cmake -C Release -V -O test.log
        If the defaults need change, execute from this directory:
            ctest -S HDF4_Examples.cmake,CTEST_SOURCE_NAME=MyExamples,INSTALLDIR=MyLocation -C Release -V -O test.log
 
-When executed, the ctest script will save the results to the log file, test.log, as
-indicated by the ctest command. If you wish the to see more build and test information,
-add "-VV" to the ctest command. The output should show near the end;
-      100% tests passed, 0 tests failed out of 49.
+    When executed, the ctest script will save the results to the log file, test.log, as
+    indicated by the ctest command. If you wish the to see more build and test information,
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply release_notes/USING_CMake_Examples.txt:71]: right.

File: release_notes/USING_HDF4_CMake.txt (hunk 1 above)
Comment (byrnHDF) [reply release_notes/USING_HDF4_CMake.txt:41]: okay

File: release_notes/USING_CMake_Examples.txt [hunk 8]
```diff
@@ -59,17 +58,19 @@ Default installation process:
...
+    add "-VV" to the ctest command. The output should show;
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply release_notes/USING_CMake_Examples.txt:72]: I'll just leave it out.

File: release_notes/USING_CMake_Examples.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (lrknox) [reply release_notes/USING_CMake_Examples.txt:66]: Do we need the missing direction?

File: release_notes/USING_CMake_Examples.txt [hunk 9]
```diff
@@ -59,17 +58,19 @@ Default installation process:
...
+           HDF4_Examples_options.cmake file.
 
        If the defaults are okay, execute from this directory:
            ctest -S HDF4_Examples.cmake -C Release -V -O test.log
        If the defaults need change, execute from this directory:
            ctest -S HDF4_Examples.cmake,CTEST_SOURCE_NAME=MyExamples,INSTALLDIR=MyLocation -C Release -V -O test.log
 
-When executed, the ctest script will save the results to the log file, test.log, as
-indicated by the ctest command. If you wish the to see more build and test information,
-add "-VV" to the ctest command. The output should show near the end;
```
[the remark this answers is not in the document]
Comment (lrknox) [reply release_notes/USING_CMake_Examples.txt:71]: Yes, but line is removed

**Review by lrknox — APPROVED** [review]

**Review by bmribler — APPROVED** [review]

**Pull request merged**